### ISSUE TITLE
Frontend Discussion

### ISSUE THREAD

**Issue by flyingant** [Wiredcraft/coding-standard-n-strategies#1]
Let's start to have a conversation here for front end developers.

Suggestions from me:

Architecture

1. React App
2. Static Web Generator (Jekyll etc.)
3. Mini Programm
4. Electron App
5. Vuejs App
6. Updating ...

Coding
1. The checklist for HTML based project (All projects). https://github.com/thedaviddias/Front-End-Checklist
2. Updating ..,.

**Comment by hunvreus** [comment]
1. For Jekyll, I've already created [`Wiredcraft/jekyll-basics`](https://github.com/Wiredcraft/jekyll-basics) that captures most of what I think are must-dos for Jekyll. I've thought about creating a Jekyll guide for a while and give it away as a free ebook. It could coincide with releasing JekyllPro.
2. I've tried Vue.js and prefer it to React, mostly because it has a smoother learning curve, but I thought the team had more or less standardized everything around React..?

**Comment by yiwan** [comment]
For website, React may overhead, we can use some thing more light weight(without the build/dev setting, e.g. Webpack or other monster tools).

For web-app, React is still good, or we can try others, like Ember(which provides almost every thing needs for creating web-app).

**Comment by flyingant** [comment]
@yiwan Agreed. Maybe you could start to try to create a simple guideline for how to use the framework as you said.

**Comment by hunvreus** [comment]
I'd rather we don't venture into other frameworks (e.g. Vue or Ember) just yet. Re-assessing our tool-belt and best practices for React though would be good.

**Comment by z4rd** [comment]
Best practices +1

**Comment by ole3021** [comment]
@flyingant @hunvreus  for the GitHub static pages, I know something cool [here](https://github.com/rafrex/spa-github-pages), which is allow people write pure React(No matter which framework you use.) to build the static pages that works in GitHub static pages. 

In case you want to know :)
